java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -417,4 +422,9 @@ public SymbolicValue exitValue() {
   public boolean exitingOnRuntimeException() {
     return exitSymbolicValue instanceof SymbolicValue.ExceptionalSymbolicValue && ((SymbolicValue.ExceptionalSymbolicValue) exitSymbolicValue).exceptionType() == null;
   }
+
+  public boolean stackIsEmpty() {
```

Comment:
I'm not convinced by the interest of this one method. You are basically creating it only to use it in one place. See other comment.